pkg/tests/end_to_end_tests/ingest_trace_test.go

```diff
@@ -79,9 +80,65 @@ func TestQueryTraces(t *testing.T) {
 		getOperationsTest(t, q)
 		findTraceTest(t, q)
 		getDependenciesTest(t, q)
+		findTracePlanTest(t, q, db)
 	})
 }
 
+func findTracePlanTest(t testing.TB, q *store.Store, db *pgxpool.Pool) {
```

Comment:
Seeing this test here makes me believe that it's more appropriate to generate SQL in the extension and have it tested there... We already moved some pure SQL things into the extension.

Reply:
+1, What is the contract defined between connector and extension?

Reply:
While re-writing the logic in a database function may have many benefits, I think that's a huge change that is beyond the scope of this PR.